### ISSUE TITLE
Training a brain converges to producing NaNs causing the agents to do nothing

### ISSUE THREAD

**Issue by ConorZAM** [Unity-Technologies/ml-agents#89]
I'm attempting to train a brain to jump, however it converges on a solution of all NaNs in the act array, I am passing it 5 states [0.25f, 0.0f, 0.0f, 0.0f, 0.0f] initially. I wouldn't imagine the inputs to be a problem yet the whole thing collapses.
If you need anything further to investigate the issue please let me know, I really want to get this working!

EDIT: This appears to be a common issue in deep learning, I'm no expert but I gather it's to do with gradients becoming very small and floating point precision making them zero. Anyway, to fix it you're best off decreasing the value of the learning rate.

**Comment by awjuliani** [comment]
Hi @ConorZAM,

Sorry you are having this issue. As you mention, large learning rates can cause `NaN` or `Inf` values to appear. It is also possible when there is division or `log()` operations with too-small numbers. This is often solved by adding a small value (typically something like `1e-10`) to the input to these operations for numerical stability.
